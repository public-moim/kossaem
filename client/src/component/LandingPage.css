.landingContainer{
    height: 100vh;
    background-size: 700px;
    background-repeat: no-repeat;
    background-position-x:right;
    
    background-color: var(--sky-blue);
    padding-left: 70px;
    padding-right: 70px;
}

.header{
    font-size: 90px;
    letter-spacing: 3px;
    word-spacing: 3px;
    font-weight: 600;
    width: 850px;
    
    padding-top: 150px;
    
}

.slogan{
    color: var(--dark-blue);
    
    font-size: 32px;
    font-weight: 600;
}

.orange-text{
    color: var(--dark-orange);
}

@media screen and (max-width:1400px) {
    .landingContainer{
        background-position-x: 140%;

    }
    
    .header{
        width: 700px;
        
        font-size: 70px;
    }

    .slogan{
        font-size: 20px;
    }
}

@media screen and (max-width:900px) {
    .landingContainer{
        background-position-y: 400px;
        display: flex;
        flex-direction: column;
        align-items: center;

    }
    
    .header{
        width: 600px;
        
        font-size: 60px;
        text-align: center;
        padding-top: 10vh;
    }

    .slogan{
        font-size: 20px;
       
        text-align: center;
    }

}

@media screen and (max-width:650px) {
    .landingContainer{
        background-position-y: 100px;
        display: flex;
        flex-direction: column;
        align-items: center;
        padding:0 20px;
        background-size: 500px;

    }
    
    .header{
        width: 100%;
        letter-spacing: 0px;
    word-spacing: 0px;
        font-size: 30px;
        text-align: center;
        padding-top: 20vh;
        
    }

    .slogan{
        font-size: 15px;
       
        text-align: center;
    }

   

}
