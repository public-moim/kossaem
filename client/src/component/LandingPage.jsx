import React from 'react'
import './LandingPage.css'
import bg from '../bg-1.png'
import { Link } from 'react-router-dom'

function LandingPage() {
  return (
    <div className='landingContainer' >
      <div className="header">
      Best Choice to 
        Learn and Test 
        your Korean Skill
      </div>
      <p className='slogan'>Solve <span className='orange-text'>Quiz</span> and Earn <span className='orange-text'>Points</span></p>
      <div className="flex">
      <Link to='/signup/student'><div className="button-primary">Get Started </div></Link>
      <Link className='link' to='/signup/teacher'><div className="button-transparent">Become a Teacher </div></Link>
    
      </div>
      
    </div>
  )
}

export default LandingPage
