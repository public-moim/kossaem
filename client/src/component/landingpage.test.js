import { fireEvent, render, screen } from '@testing-library/react';
import LandingPage from './LandingPage';
import { MemoryRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event'

test('renders Landing Page component correctly', async ()=> {
    const user= userEvent.setup()

    //render component
    const {container}=render(<Router history={history}><LandingPage/></Router>)
    expect(container).toMatchSnapshot();

    
})


