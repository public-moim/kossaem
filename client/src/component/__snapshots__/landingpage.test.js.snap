// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Landing Page component correctly 1`] = `
<div>
  <div
    class="landingContainer"
  >
    <div
      class="header"
    >
      Best Choice to Learn and Test your Korean Skill.
    </div>
    <p
      class="slogan"
    >
      Solve 
      <span
        class="orange-text"
      >
        Quiz
      </span>
       and Earn 
      <span
        class="orange-text"
      >
        Points
      </span>
    </p>
    <div
      class="flex"
    >
      <a
        href="/signup/student"
      >
        <div
          class="button-primary"
        >
          Get Started 
        </div>
      </a>
      <a
        class="link"
        href="/signup/teacher"
      >
        <div
          class="button-transparent"
        >
          Become a Teacher 
        </div>
      </a>
    </div>
  </div>
</div>
`;
